crates/solver/src/liquidity/zeroex.rs

```diff
@@ -77,6 +93,40 @@ impl ZeroExLiquidity {
         };
         Some(Liquidity::LimitOrder(limit_order))
     }
+
+    async fn run_orderbook_fetching(
+        api: Arc<dyn ZeroExApi>,
+        blocks_stream: CurrentBlockStream,
+        orderbook_cache: Arc<OrderbookCache>,
+        gpv2_address: H160,
+    ) {
+        let mut block_stream = into_stream(blocks_stream);
+        while block_stream.next().await.is_some() {
+            let queries = &[
+                // orders fillable by anyone
+                OrdersQuery::default(),
+                // orders fillable only by our settlement contract
+                OrdersQuery {
+                    sender: Some(gpv2_address),
+                    ..Default::default()
+                },
+            ];
+            let zeroex_orders_results =
+                futures::future::join_all(queries.iter().map(|query| api.get_orders(query))).await;
+            let zeroex_orders = zeroex_orders_results
+                .into_iter()
+                .flat_map(|result| {
+                    result.unwrap_or_else(|err| {
+                        tracing::error!(?err, "ZeroExResponse error during liqudity fetching");
+                        vec![]
+                    })
+                })
+                .collect();
+
+            let mut cache = orderbook_cache.write().await;
+            *cache = zeroex_orders;
```

Comment:
Currently the cache simply replaces whatever we have cached at the moment.
This could even be done without a `RwLock` by using [arc_swap](https://docs.rs/arc-swap/latest/arc_swap/type.ArcSwap.html#method.swap).
That way you can atomically swap the pointer to the old state with a pointer to the new state and don't incur any synchronization overhead when reading and don't block readers while updating the cache.